### DISCUSSION TITLE
I would like to build v2ray plug in for iOS

### DISCUSSION THREAD

**Discussion by Bamboo742** [v2fly/v2ray-core#3171] [Q&A] [answered]
I couldn't find how to integrate v2ray for ios I have a vpn app for android and I'm able to connect as v2ray client but could not find any library or anything on ios

Can anyone help me with that how can I create a client for ios

**Comment by Bamboo742** [comment]
I am now extending this flutter_v2ray plugin https://github.com/blueboy-tm/flutter_v2ray/

**Comment by Bamboo742** [comment]
I have compiled the .a file, but need .h. How should I write this?

**Comment by hubeoo** [comment]
You can learn this [Flutter调用Go的库文件](https://juejin.cn/post/7276269179073200183)

**Comment by hubeoo** [comment, marked as answer]
```
CC=~/go/misc/ios/clangwrap.sh GOOS=ios GOARCH=arm64 CGO_ENABLED=1 
go build -buildmode=c-archive -o out/ios/ios/libnativeffi.a
```
If you don't use `-buildmode=c-archive`, Go won't automatically generate `.h` files for you

**Comment by komfcore** [comment]
Did you finish with making iOS plugin?
